### ISSUE TITLE
The config that worked for me, could help

### ISSUE THREAD

**Issue by B-Lounes** [maitrix-org/Voila#20]
## Summary

This PR fixes several compatibility issues that prevent `infer.py` from running with recent versions of `transformers`, `tokenizers`, and `torch`. All changes have been tested and confirmed working on an NVIDIA GH200 (CUDA 12.4).

## Problems

1. **`DynamicCache.seen_tokens` removed in newer `transformers`** — The `seen_tokens` attribute was replaced by `get_seq_length()`, causing an `AttributeError` during generation.

2. **Deprecated `use_flash_attention_2` parameter** — Newer `transformers` versions require `attn_implementation="flash_attention_2"` instead of the legacy `use_flash_attention_2=True` kwarg.

3. **`torch.multinomial` crash with invalid probability distributions** — The logits fed into softmax can contain `inf`/`nan` values, producing invalid probability tensors that cause a CUDA-side assertion failure: `probability tensor contains either 'inf', 'nan' or element < 0`.

4. **Unpinned dependencies in `requirements.txt`** — Without version pins, `pip install` pulls the latest versions which are incompatible with the codebase.

### `model.py`
- Replaced `past_key_values.seen_tokens` with `past_key_values.get_seq_length()` in all 3 model classes (`VoilaModel`, `VoilaAudioAlphaModel`, `VoilaAutonomousModel`).

### `infer.py`
- Changed `use_flash_attention_2=True` to `attn_implementation="flash_attention_2"` in `load_model()`.

### `audio_transformer.py`
- Added logit clamping and NaN sanitization before softmax sampling to prevent `torch.multinomial` from receiving invalid probability distributions.

### `requirements.txt`
- Pinned all dependencies to exact tested versions:

```
torch==2.5.1
torchvision==0.20.1
torchaudio==2.5.1
transformers==4.46.3
tokenizers==0.20.4
flash-attn==2.8.3
safetensors==0.7.0
huggingface_hub==0.36.2
soundfile==0.13.1
librosa==0.11.0
jsonlines==4.0.0
gradio==6.5.1
pyannote.audio==3.4.0
einops==0.8.2
sentencepiece==0.2.1
speechbrain==1.0.3
lightning==2.6.1
sympy==1.13.1
numpy==2.2.6
```

## Tested Environment

| Component | Version |
|-----------|---------|
| Python | 3.10.12 |
| CUDA | 12.4 |
| GPU | NVIDIA GH200 144G HBM3e |
| flash-attn | 2.8.3 |
| transformers | 4.46.3 |
| torch | 2.5.1 |
